pkg/uploadserver/uploadserver.go

```diff
@@ -457,14 +460,21 @@ func filesystemCloneProcessor(stream io.ReadCloser, destDir string) error {
 		return errors.Wrapf(err, "error removing contents of %s", destDir)
 	}
 
-	gzr, err := gzip.NewReader(stream)
-	if err != nil {
-		return errors.Wrap(err, "error creating gzip reader")
-	}
-
-	if err = util.UnArchiveTar(gzr, destDir); err != nil {
+	if err := util.UnArchiveTar(newSnappyReadCloser(stream), destDir); err != nil {
 		return errors.Wrapf(err, "error unarchiving to %s", destDir)
 	}
 
 	return nil
 }
+
+func newContentReader(stream io.ReadCloser, contentType string) io.ReadCloser {
+	if contentType == BlockdeviceClone {
```

Comment:
I'm finding it strange that we need this conditional but the old code didn't

Reply:
The data processor code can handle gzipped streams but not snappy compressed streams